src/Raven.Server/Utils/CpuUsage.cs

```diff
@@ -18,11 +18,24 @@ public static class CpuUsage
     {
         private static readonly Logger Logger = LoggingSource.Instance.GetLogger<MachineResources>("Server");
         private static readonly object Locker = new object();
+        private static ExtensionPointInfo _previousExtensionPointInfo;
         private static WindowsInfo _previousWindowsInfo;
         private static LinuxInfo _previousLinuxInfo;
         private static MacInfo _previousMacInfo;
         private static (double MachineCpuUsage, double ProcessCpuUsage)? _lastCpuInfo;
         private static readonly (double MachineCpuUsage, double ProcessCpuUsage) EmptyCpuUsage = (0, 0);
+        private static CpuUsageExtensionPoint _cpuUsageExtensionPoint;
+
+
+        public static CpuUsageExtensionPoint CpuUsageExtensionPoint
+        {
+            get => _cpuUsageExtensionPoint;
+            set
+            {
+                _previousExtensionPointInfo = new ExtensionPointInfo(value.Data);
```

Comment:
What do we do with the previous value, if we have one, and if we don't, then it should be just a Lazy value?

Reply:
@maximburyak The `CpuUsageExtensionPoint` should be set once in the server initialization. Maybe if I will pass it to a function it will make more sense?

Reply:
Maybe the CpuUsage class shouldn't be a "static" class, but a singleton? And this way you'll initialize all the values you need inside the constructor? It looks to me that it lacks structure and state this way..